documentation/src/components/side-nav-search.ts

```diff
@@ -0,0 +1,212 @@
+        }
+        return result;
+    });
+}
+
+interface Result {
+    name: string;
+    url: string;
+}
+
+interface ResultGroup {
+    name: string;
+    results: Result[];
+    maxScore: number;
+}
+
+class SearchComponent extends LitElement {
+    private index: lunr.Index | undefined;
+    private closeOverlay?: () => void;
+
+    @query('sp-popover')
+    private popover!: HTMLElement;
+
+    public static get styles(): CSSResultArray {
+        return [sideNavSearchMenuStyles];
+    }
+
+    @property({ type: Array })
+    public results: ResultGroup[] = [];
+
+    private handleSearchInput(event: InputEvent) {
+        if (event.target) {
+            const searchField = event.target as Search;
+            this.updateSearchResults(searchField.value);
+        }
+    }
+
+    private openPopover() {
+        if (!this.popover) return;
+
+        this.closeOverlay = Overlay.open(this, 'click', this.popover, {
+            placement: 'bottom',
+        });
+    }
+
+    private closePopover() {
+        if (this.closeOverlay) {
+            this.closeOverlay();
+            delete this.closeOverlay;
+        }
+    }
+
+    private updateSearchResults(value: string): void {
+        if (!this.index || value.length < 3) {
+            this.closePopover();
+            return;
+        }
+
+        const searchParam = `${value.trim()} ${value.trim()}*`;
+
+        const collatedResults = new Map<
+            string,
+            {
+                maxScore: number;
+                results: Result[];
+            }
+        >();
+
+        const search = this.index.search(searchParam);
+        for (const item of search) {
+            const match = /^\/([^/]+)\/([^/]+)/.exec(item.ref);
+            if (!match) continue;
+
+            const [, category, name] = match;
+            if (!collatedResults.has(category)) {
+                collatedResults.set(category, {
+                    maxScore: 0,
+                    results: [],
+                });
+            }
+            const catagoryData = collatedResults.get(category);
+            if (catagoryData) {
+                catagoryData.maxScore = Math.max(
+                    catagoryData.maxScore,
+                    item.score
+                );
+                catagoryData.results.push({
+                    name: nameToTitle(name),
+                    url: item.ref,
+                });
+            }
+        }
+
+        const result: ResultGroup[] = [];
+        for (const [name, { results, maxScore }] of collatedResults) {
+            result.push({ name, results, maxScore });
+        }
+        result.sort((a, b) => {
+            if (a.maxScore < b.maxScore) return 1;
+            if (a.maxScore > b.maxScore) return -1;
+            return 0;
+        });
+        this.results = result;
+
+        this.openPopover();
+    }
+
+    firstUpdated(): void {
+        import('../../components/searchIndex.json').then((searchIndex) => {
```

Comment:
I think because this is `import`ed instead of `fetch`ed it is pushed into the initial bundle. It's kind of large, and would be nice to defer. However, it also has the nicer component name listing, so it may also be nice to use more broadly... see related note in https://github.com/adobe/spectrum-web-components/pull/312/files#diff-e9f9b198c8c59ad0fe92c58f32b4f33b

Reply:
@Westbrook  Actually, using `import()` in webpack results in a separate dynamically loaded js file. So that is not a problem here. I can see about using this data more broadly though.

<img width="1068" alt="Screen Shot 2020-02-06 at 9 26 34 AM" src="https://user-images.githubusercontent.com/154097/73962227-e28d7c80-48c2-11ea-8e9d-d49ff2e10bdc.png">

Reply:
Oh, it parses it into a JS file, got it. I was like "where's the .json file" 🙈

Reply:
Ok, I reworked things so that we always work off of the search index, even for side bar.